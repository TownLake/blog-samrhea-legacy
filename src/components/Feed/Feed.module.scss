@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.feed {
  .item {
    @include margin-bottom(1.25);

    &:last-child {
      @include margin-bottom(0.5);
    }

    .title {
      font-size: $typographic-base-font-size * 1.6875;
      @include line-height(1.5);
      @include margin-top(0);
      @include margin-bottom(0.5);

      .link {
        color: var(--color-typographic-base-font);

        &:hover,
        &:focus {
          border-bottom: 1px solid $color-dark;
          color: var(--color-typographic-base-font);
        }
      }
    }

    .description {
      font-size: $typographic-base-font-size;
      @include line-height(1);
      @include margin-bottom(0.75);
    }

    .meta {
      .time {
        color: var(--color-typographic-base-font);
        font-size: $typographic-small-font-size;
        font-weight: 600;
        text-transform: uppercase;
      }

      .divider {
        @include margin(0, 0.5);
      }

      .category {
        .link {
          color: var(--color-secondary);
          font-size: $typographic-small-font-size;
          font-weight: 600;
          text-transform: uppercase;

          &:hover,
          &:focus {
            color: $color-blue;
          }
        }
      }
    }

    .more {
      color: var(--color-primary);
      font-size: $typographic-base-font-size;

      &:hover,
      &:focus {
        border-bottom: 1px solid $color-blue;
        color: var(--color-primary);
      }
    }
  }
}
