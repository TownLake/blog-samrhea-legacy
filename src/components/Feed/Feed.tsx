import React from "react";

import { Link } from "gatsby";

import { Edge } from "@/types";

import * as styles from "./Feed.module.scss";

type Props = {
  edges: Array<Edge>;
};

const Feed: React.FC<Props> = ({ edges }: Props) => (
  <div className={styles.feed}>
    {edges.map((edge) => (
      <div className={styles.item} key={edge.node.fields.slug}>
        <div className={styles.meta}>
          <time
            className={styles.time}
            dateTime={new Date(edge.node.frontmatter.date).toLocaleDateString(
              "en-US",
              { year: "numeric", month: "long", day: "numeric" },
            )}
          >
            {new Date(edge.node.frontmatter.date).toLocaleDateString("en-US", {
              year: "numeric",
              month: "long",
            })}
          </time>
          <span className={styles.divider} />
          <span className={styles.category}>
            <Link to={edge.node.fields.categorySlug} className={styles.link}>
              {edge.node.frontmatter.category}
            </Link>
          </span>
        </div>
        <h2 className={styles.title}>
          <Link
            className={styles.link}
            to={edge.node.frontmatter?.slug || edge.node.fields.slug}
          >
            {edge.node.frontmatter.title}
          </Link>
        </h2>
        <p className={styles.description}>
          {edge.node.frontmatter.description}
        </p>
        <Link
          className={styles.more}
          to={edge.node.frontmatter?.slug || edge.node.fields.slug}
        >
          Read
        </Link>
      </div>
    ))}
  </div>
);

export default Feed;
