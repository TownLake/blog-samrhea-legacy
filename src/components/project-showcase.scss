.project-showcase {
    max-width: 1200px;
    margin: 0 auto;
    padding: 2rem;
  
    h1 {
      text-align: center;
      margin-bottom: 2rem;
      font-size: 2rem;
      font-weight: bold;
    }
  
    .project-grid {
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
      gap: 2rem;
    }
  }
  
  // Dark mode styles
  @media (prefers-color-scheme: dark) {
    .project-showcase {
      background-color: #1a202c;
      color: #f7fafc;
    }
  }