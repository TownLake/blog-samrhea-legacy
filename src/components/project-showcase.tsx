.project-tile {
    position: relative;
    width: 100%;
    height: 0;
    padding-bottom: 75%; // 4:3 aspect ratio
    overflow: hidden;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
  
    &:hover {
      transform: translateY(-5px);
    }
  
    .project-link {
      text-decoration: none;
      color: inherit;
    }
  
    .project-image {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-size: cover;
      background-position: center;
    }
  
    .project-overlay {
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0;
      background: rgba(0, 0, 0, 0.7);
      color: white;
      padding: 1rem;
      transform: translateY(100%);
      transition: transform 0.3s ease;
    }
  
    &:hover .project-overlay {
      transform: translateY(0);
    }
  
    .project-title {
      margin: 0 0 0.5rem;
      font-size: 1.2rem;
    }
  
    .project-description {
      margin: 0;
      font-size: 0.9rem;
      opacity: 0.8;
    }
  }