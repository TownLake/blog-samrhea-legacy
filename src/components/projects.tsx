import React from 'react';
import { Link } from 'gatsby';
import './project.scss';

interface ProjectProps {
  title: string;
  description: string;
  imageUrl: string;
  projectUrl: string;
}

const Project: React.FC<ProjectProps> = ({ title, description, imageUrl, projectUrl }) => {
  return (
    <div className="project-tile">
      <Link to={projectUrl} className="project-link">
        <div className="project-image" style={{ backgroundImage: `url(${imageUrl})` }}>
          <div className="project-overlay">
            <h3 className="project-title">{title}</h3>
            <p className="project-description">{description}</p>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default Project;