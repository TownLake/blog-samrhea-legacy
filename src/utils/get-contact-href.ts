import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
