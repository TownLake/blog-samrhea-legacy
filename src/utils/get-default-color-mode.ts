const getDefaultColorMode = (): "dark" | "light" => {
  if (typeof window === "undefined") {
    return "light";
  }

  const mql = window.matchMedia("(prefers-color-scheme: dark)");
  const hasMediaQueryPreference = typeof mql.matches !== "undefined";
  if (hasMediaQueryPreference) {
    return mql.matches ? "dark" : "light";
  }

  return "light";
};

export default getDefaultColorMode;
