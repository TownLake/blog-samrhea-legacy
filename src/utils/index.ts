export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
