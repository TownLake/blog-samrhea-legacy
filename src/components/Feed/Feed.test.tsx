import React from "react";

import { Feed } from "@/components/Feed";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Feed", () => {
  test("renders correctly", () => {
    const props = { edges: mocks.edges };
    const tree = testUtils
      .createSnapshotsRenderer(<Feed {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
