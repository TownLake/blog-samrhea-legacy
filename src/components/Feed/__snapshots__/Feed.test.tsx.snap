// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div>
  <div>
    <div>
      <time
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span />
      <span>
        <a
          href="/typography"
        >
          Typography
        </a>
      </span>
    </div>
    <h2>
      <a
        href="/posts/perfecting-the-art-of-perfection"
      >
        Perfecting the Art of Perfection
      </a>
    </h2>
    <p>
      An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.
    </p>
    <a
      href="/posts/perfecting-the-art-of-perfection"
    >
      Read
    </a>
  </div>
  <div>
    <div>
      <time
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span />
      <span>
        <a
          href="/design-inspiration"
        >
          Design Inspiration
        </a>
      </span>
    </div>
    <h2>
      <a
        href="/posts/johannes-gutenberg"
      >
        Johannes Gutenberg: The Birth of Movable Type
      </a>
    </h2>
    <p>
      German inventor Johannes Gutenberg developed a method of movable type and used it to create one of the western world’s first major printed books, the “Forty–Two–Line” Bible.
    </p>
    <a
      href="/posts/johannes-gutenberg"
    >
      Read
    </a>
  </div>
  <div>
    <div>
      <time
        dateTime="September 1, 2017"
      >
        September 2017
      </time>
      <span />
      <span>
        <a
          href="/typography"
        >
          Typography
        </a>
      </span>
    </div>
    <h2>
      <a
        href="/posts/humane-typography-in-the-digital-age"
      >
        Humane Typography in the Digital Age
      </a>
    </h2>
    <p />
    <a
      href="/posts/humane-typography-in-the-digital-age"
    >
      Read
    </a>
  </div>
</div>
`;
